src/Path.php

```diff
@@ -27,13 +27,22 @@ public function relativize(string $absolutePath): string
     {
         $relativePath = str_replace($this->workingDirectory, '.', $absolutePath);
 
-        return is_dir($absolutePath) ? rtrim($relativePath, '/').'/' : $relativePath;
+        return is_dir($absolutePath) ? rtrim($relativePath, DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR : $relativePath;
     }
 
     public function concatenate(array $parts): string
     {
         return array_reduce($parts, function (string $initial, string $next): string {
-            return rtrim($initial, '/').'/'.ltrim($next, '/');
+            if (strlen($initial) === 0) {
+                return $this->normalize($next);
+            }
+
+            return rtrim($this->normalize($initial), DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR.ltrim($this->normalize($next), DIRECTORY_SEPARATOR);
         }, '');
     }
+
+    private function normalize(string $path): string
+    {
+        return str_replace(['/', '\\'], DIRECTORY_SEPARATOR, $path);
```

Comment:
This should be `str_replace('/', DIRECTORY_SEPARATOR, $path);`: `\` should not be replaced on non-Windows.
BUT, I'm not sure we need this normalization at all. That's not the job of the concatenation method.